File: examples/javascript/web3-integration/wallet.js
/**
 * JavaScript Web3 Wallet Integration
 * Modern Web3.js example for Ethereum interaction
 */

const { Web3 } = require('web3');
const fs = require('fs');

class Web3Wallet {
    constructor(rpcUrl = 'https://eth-mainnet.g.alchemy.com/v2/YOUR-API-KEY') {
        this.web3 = new Web3(rpcUrl);
    }

    /**
     * Get ETH balance for an address
     */
    async getBalance(address) {
        const balance = await this.web3.eth.getBalance(address);
        return this.web3.utils.fromWei(balance, 'ether');
    }

    /**
     * Send ETH transaction
     */
    async sendTransaction(from, to, amount, privateKey) {
        const account = this.web3.eth.accounts.privateKeyToAccount(privateKey);
        this.web3.eth.accounts.wallet.add(account);

        const tx = {
            from: from,
            to: to,
            value: this.web3.utils.toWei(amount, 'ether'),
            gas: 21000,
        };

        const receipt = await this.web3.eth.sendTransaction(tx);
        return receipt;
    }

    /**
     * Interact with ERC-20 token
     */
    async getTokenBalance(tokenAddress, walletAddress) {
        const minABI = [
            {
                constant: true,
                inputs: [{ name: '_owner', type: 'address' }],
                name: 'balanceOf',
                outputs: [{ name: 'balance', type: 'uint256' }],
                type: 'function',
            },
            {
                constant: true,
                inputs: [],
                name: 'decimals',
                outputs: [{ name: '', type: 'uint8' }],
                type: 'function',
            },
        ];

        const contract = new this.web3.eth.Contract(minABI, tokenAddress);
        const balance = await contract.methods.balanceOf(walletAddress).call();
        const decimals = await contract.methods.decimals().call();

        return Number(balance) / 10 ** Number(decimals);
    }

    /**
     * Transfer ERC-20 tokens
     */
    async transferToken(tokenAddress, from, to, amount, privateKey) {
        const account = this.web3.eth.accounts.privateKeyToAccount(privateKey);
        this.web3.eth.accounts.wallet.add(account);

        const minABI = [
            {
                constant: false,
                inputs: [
                    { name: '_to', type: 'address' },
                    { name: '_value', type: 'uint256' },
                ],
                name: 'transfer',
                outputs: [{ name: '', type: 'bool' }],
                type: 'function',
            },
        ];

        const contract = new this.web3.eth.Contract(minABI, tokenAddress);
        const tx = contract.methods.transfer(to, amount);

        const gas = await tx.estimateGas({ from: from });
        const receipt = await tx.send({ from: from, gas: gas });

        return receipt;
    }

    /**
     * Sign a message
     */
    async signMessage(message, privateKey) {
        const account = this.web3.eth.accounts.privateKeyToAccount(privateKey);
        const signature = account.sign(message);
        return signature;
    }

    /**
     * Verify signature
     */
    recoverSigner(message, signature) {
        return this.web3.eth.accounts.recover(message, signature);
    }

    /**
     * Get current gas price
     */
    async getGasPrice() {
        const gasPrice = await this.web3.eth.getGasPrice();
        return this.web3.utils.fromWei(gasPrice, 'gwei');
    }

    /**
     * Get latest block number
     */
    async getBlockNumber() {
        return await this.web3.eth.getBlockNumber();
    }

    /**
     * Listen for new blocks
     */
    subscribeToBlocks(callback) {
        const subscription = this.web3.eth.subscribe('newBlockHeaders');
        subscription.on('data', callback);
        return subscription;
    }
}

// Example usage
async function main() {
    const wallet = new Web3Wallet();

    // Get balance
    const balance = await wallet.getBalance('0x742d35Cc6634C0532925a3b844Bc9e7595f0bEb');
    console.log(`Balance: ${balance} ETH`);

    // Get gas price
    const gasPrice = await wallet.getGasPrice();
    console.log(`Gas Price: ${gasPrice} Gwei`);

    // Get block number
    const blockNumber = await wallet.getBlockNumber();
    console.log(`Latest Block: ${blockNumber}`);
}

if (require.main === module) {
    main().catch(console.error);
}

module.exports = Web3Wallet;

File: examples/dart/web3-example/lib/wallet.dart
// Dart Web3 Wallet for Flutter
import 'package:web3dart/web3dart.dart';
import 'package:http/http.dart';

class Web3Wallet {
  late Web3Client client;
  final String rpcUrl;

  Web3Wallet(this.rpcUrl) {
    client = Web3Client(rpcUrl, Client());
  }

  Future<EtherAmount> getBalance(String address) async {
    final ethAddress = EthereumAddress.fromHex(address);
    return await client.getBalance(ethAddress);
  }

  Future<int> getBlockNumber() async {
    return await client.getBlockNumber();
  }

  Future<EtherAmount> getGasPrice() async {
    return await client.getGasPrice();
  }

  Future<String> sendTransaction({
    required String privateKey,
    required String toAddress,
    required double amountEther,
  }) async {
    final credentials = EthPrivateKey.fromHex(privateKey);
    final to = EthereumAddress.fromHex(toAddress);
    final amount = EtherAmount.fromUnitAndValue(
      EtherUnit.ether,
      amountEther,
    );

    final txHash = await client.sendTransaction(
      credentials,
      Transaction(
        to: to,
        value: amount,
      ),
      chainId: 1,
    );

    return txHash;
  }

  void dispose() {
    client.dispose();
  }
}

void main() async {
  final wallet = Web3Wallet('https://eth-mainnet.g.alchemy.com/v2/YOUR-API-KEY');

  final vitalik = '0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045';

  final balance = await wallet.getBalance(vitalik);
  final blockNumber = await wallet.getBlockNumber();
  final gasPrice = await wallet.getGasPrice();

  print('Balance: ${balance.getValueInUnit(EtherUnit.ether)} ETH');
  print('Block: $blockNumber');
  print('Gas: ${gasPrice.getValueInUnit(EtherUnit.gwei)} Gwei');

  wallet.dispose();
}
